backend/controller/ingress/ingress_test.go

```diff
@@ -73,3 +74,54 @@ func TestValidation(t *testing.T) {
 		})
 	}
 }
+
+func TestParseQueryParams(t *testing.T) {
+	tests := []struct {
+		query   string
+		request obj
+		err     string
+	}{
+		{query: "", request: obj{}},
+		{query: "a=10", request: obj{"a": "10"}},
```

Comment:
If the schema for `a` is `[Int]`, won't this fail later when it tries to validate, because it's a scalar?

Reply:
yeah I assume it will fail later if that's the case. This function just builds the request from the query it gets. The validator will run after the map is built.
```go
func ValidateAndExtractBody(route *dal.IngressRoute, r *http.Request, sch *schema.Schema) ([]byte, error) {

// this will build the map from the query params (or url or body)
	requestMap, err := buildRequestMap(route, r)
	if err != nil {
		return nil, err
	}

// other code

       err = validateRequestMap(dataRef, []string{dataRef.String()}, requestMap, sch)
````